Samples/StyleTransfer/VideoEffect/StyleTransferEffectComponent/StyleTransferVideoEffect.cs

```diff
@@ -0,0 +1,236 @@
+                _session = Session;
+                _inputImageDescription = InputImageDescription;
+                _outputImageDescription = OutputImageDescription;
+
+                Task.Run(async () =>
+                {
+                    VideoFrame inputTransformed = await CenterCropImageAsync(inputVideoFrame, 720, 720);
+                    VideoFrame outputTransformed = new VideoFrame(BitmapPixelFormat.Bgra8, 720, 720);
+
+                    _binding.Bind(_inputImageDescription, ImageFeatureValue.CreateFromVideoFrame(inputTransformed));
+                    _binding.Bind(_outputImageDescription, ImageFeatureValue.CreateFromVideoFrame(outputTransformed));
+
+                    var results = _session.Evaluate(_binding, "test");
+                    await outputTransformed.CopyToAsync(outputVideoFrame);
+                }).ConfigureAwait(false).GetAwaiter().GetResult();
+            }
+        }
+
+        /// <summary>
+        /// Crop image given a target width and height 
+        /// </summary>
+        /// <param name="inputVideoFrame"></param>
+        /// <returns></returns>
+        public static IAsyncOperation<VideoFrame> CenterCropImageAsync(VideoFrame inputVideoFrame, uint targetWidth, uint targetHeight)
+        {
+            return AsyncInfo.Run(async (token) =>
+            {
+                bool useDX = inputVideoFrame.SoftwareBitmap == null;
+                VideoFrame result = null;
+                // Center crop
+                try
+                {
+
+                    // Since we will be center-cropping the image, figure which dimension has to be clipped
+                    var frameHeight = useDX ? inputVideoFrame.Direct3DSurface.Description.Height : inputVideoFrame.SoftwareBitmap.PixelHeight;
+                    var frameWidth = useDX ? inputVideoFrame.Direct3DSurface.Description.Width : inputVideoFrame.SoftwareBitmap.PixelWidth;
+
+                    Rect cropRect = GetCropRect(frameWidth, frameHeight, targetWidth, targetHeight);
+                    BitmapBounds cropBounds = new BitmapBounds()
+                    {
+                        Width = (uint)cropRect.Width,
+                        Height = (uint)cropRect.Height,
+                        X = (uint)cropRect.X,
+                        Y = (uint)cropRect.Y
+                    };
+
+                    // Create the VideoFrame to be bound as input for evaluation
+                    if (useDX)
+                    {
+                        if (inputVideoFrame.Direct3DSurface == null)
+                        {
+                            throw (new Exception("Invalid VideoFrame without SoftwareBitmap nor D3DSurface"));
+                        }
+
+                        result = new VideoFrame(BitmapPixelFormat.Bgra8,
+                                                (int)targetWidth,
+                                                (int)targetHeight,
+                                                BitmapAlphaMode.Premultiplied);

```

Comment:
this is the same in both branches? move out?

Reply:
the C# version is all out of date, it might be better to just delete it